### ISSUE TITLE
Shader compilation failure if csm_Feature used is not compatible with base material

### ISSUE THREAD

**Issue by chasedavis** [FarazzShaikh/THREE-CustomShaderMaterial#36]
Certain csm features like emissive, metalness, roughness, ao, bump only work with certain materials, makes sense!

It'd be great if rather than a shader compilation issue, there was graceful failure when those features are attempted to be used with an incompatible base material. A big use case for this would be for building material editors or otherwise offering some configuration options for materials as is popular.

I believe it can be worked around through patchmaps but might also be feasible or easy to consider **doing nothing** with the variable assignment (meaning, it's still declared in CSM compilation).

Thoughts?

<img width="445" alt="Screen Shot 2023-04-18 at 11 28 02 PM" src="https://user-images.githubusercontent.com/10068755/232959605-083c42cc-eaa5-47ae-b95a-f96b0935fc1f.png">

**Comment by FarazzShaikh** [comment]
Good idea! Would be trivial to implement as we already store the base material `type`. 

However, custom materials almost never set the `type` property to something accurate so this might only be possible to check for in first degree extensions which is completely okay

will add this to the next release

**Comment by chasedavis** [comment]
Wow so fast :D
